crates/editor/src/display_map.rs

```diff
@@ -984,7 +984,18 @@ impl DisplaySnapshot {
                 }
             }
             let end = end.unwrap_or(max_point);
-            Some((start..end, self.fold_placeholder.clone()))
+
+            // Adjust end point to exclude folding line break if present
+            let adjusted_end = if self.buffer_snapshot.is_line_blank(MultiBufferRow(end.row)) {
```

Comment:
While this works for a single blank line, it does not work for multiple blank lines, e.g. this code:

```python
def py():
    a = 4

    b = 4


def test():
    pass
```

will still miss a line break when it is folded

<img width="197" alt="image" src="https://github.com/zed-industries/zed/assets/53836821/136ae223-7192-46e2-b9fc-b81c89af81b5">

I think this could be solved by just searching upwards until we find a non-empty line, similar too how it's done for indent guides

Reply:
> While this works for a single blank line, it does not work for multiple blank lines, e.g. this code:
> 
> ```python
> def py():
>     a = 4
> 
>     b = 4
> 
> 
> def test():
>     pass
> ```
> 
> will still miss a line break when it is folded
> 
> <img alt="image" width="197" src="https://private-user-images.githubusercontent.com/53836821/340294634-136ae223-7192-46e2-b9fc-b81c89af81b5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg2MjU0MjMsIm5iZiI6MTcxODYyNTEyMywicGF0aCI6Ii81MzgzNjgyMS8zNDAyOTQ2MzQtMTM2YWUyMjMtNzE5Mi00NmUyLWI5ZmMtYjgxYzg5YWY4MWI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE3VDExNTIwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRhZGRlN2I3N2Q0MjNkZjgxNzJlNTJhODM5NzQ3MTI0MzUyOWMxNGM0MTQ3MmUwZjUzNjc3YjZlNGI2NWRmN2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.e2QB5-aQnUqD-L7ysHXliLlBqAP1w1RXAZXYd6-zqdY">
> I think this could be solved by just searching upwards until we find a non-empty line, similar too how it's done for indent guides

@bennetbo Thank you for the idea! I will try to get that done as well.